src/main/java/com/zendesk/maxwell/MaxwellTableCache.java

```diff
@@ -22,7 +23,7 @@ public void processEvent(Schema schema, MaxwellFilter filter, TableMapEvent even
 
 			Table tbl = db.findTable(tblName);
 
-			if ( filter != null && filter.isTableBlacklisted(tblName) )
+			if ( filter != null && filter.isTableBlacklisted(tbl.getDatabase().getName(), tblName) )
```

Comment:
Ok, I think I see now.

Still not sure how to do the integration tests, but I think the change I've just made should cover the crash cases you've seen.